### ISSUE TITLE
Syntax highlight colors are becoming too dark

### ISSUE THREAD

**Issue by montoner0** [wmjordan/Codist#323]
For some reason some bright colors I've configured are becoming much darker after VS restart (not every time, can't catch the exact moment). I wonder why Codist would ever tamper with the user configured colors... Also noticed that they are also darken within an exported theme...

**Comment by montoner0** [comment]
Even weirder thing: colors in preview and in a picker don't match
![image](https://github.com/wmjordan/Codist/assets/13542804/22421b94-0bc1-4d8d-8448-7c0d324fa168)

**Comment by wmjordan** [comment]
Please upload your VS theme and Codist configuration file here so I can test it.

**Comment by montoner0** [comment]
I use the standard Blue theme with non-standard colors. Please see the attached files.
[Exported-2024-04-12_colors.vssettings.zip](https://github.com/wmjordan/Codist/files/14956043/Exported-2024-04-12_colors.vssettings.zip)
[Codist.json.zip](https://github.com/wmjordan/Codist/files/14956115/Codist.json.zip)

**Comment by wmjordan** [comment]
I analyzed the VS settings and found the cause of your issue.
Codist automatically adjusts the brightness of the syntax theme to the VS theme.
However, the algorithm is not stable when the editor is dark but other text windows are bright, or vise versa.
I currently can not think of a way to perfectly deal with situations like this.
I will let you know if there is a new beta version which address this issue.

**Comment by montoner0** [comment]
As a quick solution you could add an option to disable the auto-adjusting.

**Comment by wmjordan** [comment]
Please try [beta 9822](https://github.com/wmjordan/Codist/issues/316).

**Comment by montoner0** [comment]
I was using the beta for several days and so far it seems good. However I've seen other issues (not related to highlighting) though haven't found a way to reproduce, they seem random so far...

**Comment by wmjordan** [comment]
Well.
If you encounter that problem once again, please let me know.

**Comment by montoner0** [comment]
The bug has re-appeared in the version 8. It's really annoying and pops up every time I wake a PC from sleep (considering the VS is running). Please let me know how I can help to find the issue.

**Comment by wmjordan** [comment]
Please upload your VS theme and Codist configuration file here and let me try reproducing the issue.

**Comment by montoner0** [comment]
They're mostly the same as the previous https://github.com/wmjordan/Codist/issues/323#issuecomment-2051265227

**Comment by wmjordan** [comment]
Your previously exported VS settings could not be imported.

> Your settings were imported, but there were some errors.
>
> Error 1:  Fonts and Colors: The color theme specified in the imported file is not found.  Fonts & colors will not be imported.

**Comment by wmjordan** [comment]
BTW, please download and help test a new beta in #366 , which fixes some theming problems.

**Comment by wmjordan** [comment]
It looks good to me after awaking my laptop from sleep either.

**Comment by wmjordan** [comment]
I tried putting my laptop into sleep for quite a few times. The highlight was always good. I could not reproduce this issue.
When this issue occurs on your side, does it make the colors back to normal by reloading the exported Codist config file?

**Comment by montoner0** [comment]
Another observation: the colors become more... fragile after re-importing: toggling off/on the syntax highlight option in Codist corrupts them again.

**Comment by montoner0** [comment]
Yes, I have https://marketplace.visualstudio.com/items?itemName=mhoumann.CommentsPlusDev17Preview but I don't think it's relevant to the issue.

**Comment by wmjordan** [comment]
You don't have to use that extension. Codist's comment tagger already includes similar functionality.

<img width="976" height="403" alt="Image" src="https://github.com/user-attachments/assets/6e303469-85e8-4d4d-bb0d-09bcbda66cd9" />

Please try disabling that extension and try again.

**Comment by wmjordan** [comment]
I seemed to be able to reproduce this issue, which might be irrelevant to other extensions.
Please download the new beta and see whether this issue persists.

**Comment by wmjordan** [comment]
Have you downloaded the beta in #366?

**Comment by wmjordan** [comment]
Good.
Thank you for your feedback.
Please also try to modify your syntax color settings and see whether it is properly saved and restored (after restarting VS).
Let's see whether it is fixed.

**Comment by montoner0** [comment]
I think the issue could be closed. The colors don't corrupt and restoring went okay. Though I've seen another issue after restoring, I'll create another issue for it.
